app/scripts/components/common/mapbox/layers/vector-timeseries.tsx

```diff
@@ -0,0 +1,242 @@
+    [sourceParams]
+  );
+  useEffect(() => {
+    if (!date || !featuresApiEndpoint) return;
+
+    const start = userTzDate2utcString(startOfDay(date));
+    const end = userTzDate2utcString(endOfDay(date));
+
+    const tileParams = qs.stringify({
+      ...sourceParams,
+      datetime: `${start}/${end}`
+    });
+
+    const sources: Record<string, AnySourceImpl> = {
+      [id]: {
+        type: 'vector',
+        tiles: [`${featuresApiEndpoint}/tiles/{z}/{x}/{y}?${tileParams}`]
+      } as VectorSourceImpl
+    };
+
+    const layers = [
+      {
+        id: `${id}-line-bg`,
+        type: 'line',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'line-color': theme.color?.['base-300'],
+          'line-width': 4
+        },
+        // filter: ['==', '$type', 'LineString'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      {
+        id: `${id}-line-fg`,
+        type: 'line',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'line-color': theme.color?.primary,
+          'line-width': 2
+        },
+        filter: ['==', '$type', 'LineString'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      {
+        id: `${id}-fill-fg`,
+        type: 'fill',
+        source: id,
+        'source-layer': 'default',
+        layout: {
+          visibility: isHidden ? 'none' : 'visible'
+        },
+        paint: {
+          'fill-color': theme.color?.primary,
+          'fill-opacity': 0.8
+        },
+        filter: ['==', '$type', 'Polygon'],
+        minzoom: minZoom,
+        metadata: {
+          layerOrderPosition: 'raster'
+        }
+      },
+      minZoom > 0
+        ? {
+            type: 'symbol',
+            id: `${id}-points`,
+            source: id,
+            'source-layer': 'default',
+            layout: {
+              'icon-image': 'leaflet-marker',
+              visibility: isHidden ? 'none' : 'visible',
+              'icon-offset': [0, -12]
+            },
+            maxzoom: minZoom,
+            metadata: {
+              layerOrderPosition: 'markers'
+            }
+          }
+        : undefined
+    ].filter(Boolean) as AnyLayer[];
+
+    updateStyle({
+      generatorId: 'vector-timeseries',
+      sources,
+      layers
+    });
+    // sourceParams not included, but using a stringified version of it to
+    // detect changes (haveSourceParamsChanged)
+    // `theme` will not change throughout the app use
+  }, [
+    id,
+    updateStyle,
+    date,
+    featuresApiEndpoint,
+    minZoom,
+    isHidden,
+    haveSourceParamsChanged
+  ]);
+
+  //
+  // Listen to mouse events on the markers layer
+  //
+  useEffect(() => {
```

Comment:
It would be interesting to extract that as a hook and reuse in `raster-timeseries` (provide a callback for the `onPointsClick` handler)

Reply:
Great idea!